scripts/ci-release.py

```diff
@@ -0,0 +1,37 @@
+import toml
+import requests
+import subprocess
+
+cargo_toml = toml.load("Cargo.toml")
+crate_version = cargo_toml["package"]["version"]
+print("Detected crate version " + crate_version)
+
+api_url = "https://crates.io/api/v1/crates/x86_64/versions"
+crates_io_versions = requests.get(api_url).json()
+
+new_version = True
+for version in crates_io_versions["versions"]:
+    assert (version["crate"] == "x86_64")
+    if version["num"] == crate_version:
+        new_version = False
+        break
+
+if new_version:
+    print("Could not find version " + crate_version + " on crates.io; creating a new release")
+
+    print("  Running `cargo publish`")
+    subprocess.run(["cargo", "publish"], check=True)
```

Comment:
Do we need to have a setup action to install cargo?

Reply:
I think we don't since rustup is installed by default on the Github Actions runners. Because of the `rust-toolchain` file, it automatically downloads a nightly version.